pkg/cmd/step/syntax/step_syntax_effective_test.go

```diff
@@ -490,6 +490,53 @@ func TestCreateCanonicalPipeline(t *testing.T) {
 				},
 			},
 		},
+	}, {
+		name:         "overrides-with-buildpack-using-jenkins-x-syntax",
+		pack:         "jx-syntax-in-buildpack",
+		repoName:     "jx-demo-qs",
+		organization: "abayer",
+		branch:       "master",
+		expected: &config.ProjectConfig{
```

Comment:
I have been wondering for quite some time whether for these tests it would not be better to also have the expected results as yaml directly and compare the expected yaml with the generated one. I find it hard to make sense of all these structs. It  would also move out a lot of code from this test file.  I have a feeling that it might be also easier to build the expected config in YAML directly. 

Not sure how much work this change would be. As said something I am wondering about.